### ISSUE TITLE
Audio Block: Transcript Upload Feature

### ISSUE THREAD

**Issue by ekazda** [WordPress/gutenberg#75789]
## What problem does this address?
WCAG requires transcript for pre-recorded audio. Currently the Audio block does not include a feature for uploading or providing a transcript.

## What is your proposed solution?
Add an additional fields for either uploaded transcript files or text-entry transcripts so that WCAG compliance can be achieved.

**Comment by im3dabasia** [comment]
This would be a great enhancement for the Audio block. It is a Level A requirement under WCAG 2.1, specifically for [Prerecorded Audio-only and Video-only content](https://www.w3.org/WAI/WCAG21/Understanding/audio-only-and-video-only-prerecorded.html). It’s definitely worth looking into to ensure better accessibility compliance.

**Comment by joedolson** [comment]
I'm not sure there's a meaningful way that the audio block actually requires this, unless there was support for some kind of synchronization - like a transcript navigator or visible captions.

Currently, WordPress doesn't have any support for media relationships. What would be most useful would be to have the ability to associate audio and video items in the media library with their captions and transcript files, so that the use of that media would always come along with its supporting media alternatives. 

This is an [open core issue for video](https://core.trac.wordpress.org/ticket/43178), though there's no meaningful difference considering it for audio.

**Comment by Vrishabhsk** [comment]
Hi @ekazda 👋 
- I have raised a PR to get this request started : https://github.com/WordPress/gutenberg/pull/78858
- I had initially started to work on an approach to allow user to upload a transcript but then to keep things simple, i have kept the logic to a minimum of adding the transcript manually.

### Questions
- Do you have a suggestion on how we want to handle a transcript file?

Let me know your thoughts on this. Thanks

**Comment by ekazda** [comment]
> ### Questions
> * Do you have a suggestion on how we want to handle a transcript file?

I would think the easiest way would be to simply upload a transcript file to the media gallery and link it.
